src/main/java/org/apache/fineract/api/BatchApi.java

```diff
@@ -57,10 +79,23 @@ public BatchDTO batchDetails(@RequestParam(value = "batchId", required = false)
     }
 
     @GetMapping("/batch/detail")
-    public ResponseEntity<List<Transfer>> batchDetails(@RequestParam(value = "batchId") String batchId,
+    public ResponseEntity<List<Transfer>> batchDetails(HttpServletResponse httpServletResponse,
+                                                       @RequestParam(value = "batchId") String batchId,
                                                        @RequestParam(value = "status", defaultValue = "ALL") String status,
                                                        @RequestParam(value = "pageNo", defaultValue = "0") Integer pageNo,
-                                                       @RequestParam(value = "pageSize", defaultValue = "10") Integer pageSize) {
+                                                       @RequestParam(value = "pageSize", defaultValue = "10") Integer pageSize,
+                                                       @RequestParam(value = "command", required = false, defaultValue = "json") String command) {
+
+        if (command.equalsIgnoreCase("download")) {
+            Batch batch = batchRepository.findByBatchId(batchId);
+            if (batch != null && batch.getResult_file() != null) {
+                httpServletResponse.setHeader("Location", batch.getResult_file());
+                httpServletResponse.setStatus(302);
+            } else {
+                httpServletResponse.setStatus(404);
+            }
+            return null;
```

Comment:
looks like this return is at the wrong place.

Reply:
it's at the correct place, refer to the above lines it's redirecting to the external URL.